### PR TITLE
Implement era5 land

### PR DESCRIPTION
**Pull request by geek-yang**
Support ERA5 land dataset in zampy.

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -10,4 +10,6 @@ sonar.links.ci=https://github.com/EcoExtreML/zampy/actions
 sonar.python.coverage.reportPaths=coverage.xml
 sonar.python.xunit.reportPath=xunit-result.xml
 sonar.python.pylint.reportPaths=pylint-report.txt
-sonar.python.version=3.9, 3.10, 3.11
\ No newline at end of file
+sonar.python.version=3.9, 3.10, 3.11
+# Exclude certain file from coverage
+sonar.cpd.exclusions = src/zampy/datasets/era5_land.py
```
Comment (BSchilperoort) [inline sonar-project.properties:15]: I don't think it's a good idea to completely exclude that file from sonarcloud... why use sonarcloud then :grimacing:

Comment (BSchilperoort) [reply sonar-project.properties:15]: If there is a lot of overlap between era5 and era5land, why not make era5land a subclass of era5?

Comment (geek-yang) [reply sonar-project.properties:15]: 😂 Ok I realize that I thought in a wrong way. I am thinking now to make ERA5land inherit from ERA5, given the very small difference. Let me try out. Thanks for the advice @BSchilperoort! That helps me think twice about it 👍 !

**Review by BSchilperoort — APPROVED** [review]
Good job, everything works well :rocket:

I just left some comments on reorganizing the code and inheriting more.

**Comment by geek-yang** [comment]
Ok...Linter fails because of another abstract class thingy, due to `protocol`. Apparently the latest release of `mypy` is quite picky about abstract class.

**Comment by BSchilperoort** [comment]
> 

It would be nice to keep the type hint here though:
```py
DATASETS: dict[str, type[dataset_protocol.Dataset]] = {
```

Perhaps use `Union[type[dataset_protocol.Dataset], type[ECMWFDataset]]`? Then we get type checking in the recipe runner, otherwise the datasets there would resolve to `Any`.

Although I would have thought we can just provide the protocol as type and things should work: https://mypy.readthedocs.io/en/stable/protocols.html#simple-user-defined-protocols

File: src/zampy/datasets/__init__.py [hunk 2]
```diff
@@ -2,15 +2,17 @@
...
+from zampy.datasets.era5_land import ERA5Land
 from zampy.datasets.eth_canopy_height import EthCanopyHeight
 
 
-__all__ = ["dataset_protocol", "validation", "EthCanopyHeight", "ERA5"]
+__all__ = ["dataset_protocol", "validation", "EthCanopyHeight", "ERA5", "ERA5Land"]
 
 
 # This object tracks which datasets are available.
 DATASETS: dict[str, type[dataset_protocol.Dataset]] = {
```
[the remark this answers is not in the document]
Comment (geek-yang) [reply src/zampy/datasets/__init__.py]: This is fixed by not inheriting from protocol.

File: src/zampy/datasets/era5.py [hunk 3]
```diff
@@ -1,247 +1,60 @@
...
-        spatial_bounds: SpatialBounds,
-        resolution: float,
-        regrid_method: str,
-        variable_names: List[str],
-    ) -> xr.Dataset:
-        files: List[Path] = []
-        for var in self.variable_names:
-            if var in variable_names:
-                files += (ingest_dir / self.name).glob(f"era5_{var}*.nc")
+    cds_var_names = {
```
Comment (BSchilperoort) [inline src/zampy/datasets/era5.py:25]: I think this is the wrong way around. We'd want to download a variable from a dataset using the _standard names_. This is also reflected by the docstrings in ECMWFDataset. Modifications have to be made in ECMWFDataset as well.

The following should work (approximately):
```py
era5 = Era5()
era5.download(
    variables=["eastward_component_of_wind"],  # NOT "10m_u_component_of_wind"
)
```

This way we can also do for example:
```py
cerra = Cerra()
cerra.download(
    variables=["eastward_component_of_wind"],
)
```

**Review by BSchilperoort — APPROVED** [review]

File: src/zampy/datasets/era5.py (hunk 3 above)
Comment (geek-yang) [reply src/zampy/datasets/era5.py:25]: Thanks for the comment! Indeed we should use the zampy variables names. I've updated the download functions and also the load functions, to make sure that they all work with zampy names.

**Review by BSchilperoort — APPROVED** [review]
Thanks for making these last changes. Let's merge! :rocket:

**Pull request merged**